sys/auto_init/netif/auto_init_at86rf2xx.c

```diff
@@ -58,11 +59,19 @@ void auto_init_at86rf2xx(void)
             LOG_ERROR("[auto_init_netif] error initializing at86rf2xx radio #%u\n", i);
         }
         else {
-            gnrc_netdev_init(_at86rf2xx_stacks[i],
-                             AT86RF2XX_MAC_STACKSIZE,
-                             AT86RF2XX_MAC_PRIO,
-                             "at86rf2xx",
-                             &gnrc_adpt[i]);
+#ifdef MODULE_GNRC_LWMAC
+            gnrc_lwmac_init(_at86rf2xx_stacks[i],
+                            AT86RF2XX_MAC_STACKSIZE,
+                            AT86RF2XX_MAC_PRIO,
+                            "at86rf2xx-lwmac",
+                            &gnrc_adpt[i]);
```

Comment:
This is problematic, if you have two at86rf2xx radios on one board and you want one with lwMAC and one without. Also: this doesn't scale for more than one MAC protocol. But granted: that's more a problem with our current way of auto-initializing interfaces, not with your implementation